quic/s2n-quic/src/tests.rs

```diff
@@ -291,3 +297,83 @@ fn local_stream_open_notify_test() {
     })
     .unwrap();
 }
+
+pub struct PacketSentSubscriber {
+    pub packet_sent: Arc<Mutex<Vec<PacketSent>>>,
+}
+
+pub struct PacketSentContext {
+    packet_sent: Arc<Mutex<Vec<PacketSent>>>,
+}
+
+impl Subscriber for PacketSentSubscriber {
+    type ConnectionContext = PacketSentContext;
+
+    fn create_connection_context(
+        &mut self,
+        _meta: &ConnectionMeta,
+        _info: &ConnectionInfo,
+    ) -> Self::ConnectionContext {
+        PacketSentContext {
+            packet_sent: Arc::clone(&self.packet_sent),
```

Comment:
```suggestion
            packet_sent: self.packet_sent,
```

Reply:
The compiler doesn't like that one. 

cannot move out of `self.packet_sent` which is behind a mutable reference
move occurs because `self.packet_sent` has type `Arc<...Mutex<Vec<...PacketSent>>>`, which does not implement the `Copy` trait.

Reply:
I'm dumb; I meant to add a `.clone()` to the end. Yes the event types are all non-copy in case we want to add non-copyable fields in the future